airflow/providers/mssql/hooks/mssql_odbc.py

```diff
@@ -0,0 +1,183 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module contains MS Sql Odbc hook.
+"""
+
+from urllib.parse import quote_plus
+
+import pyodbc
+
+from airflow.hooks.dbapi_hook import DbApiHook
+
+
+class MsSqlOdbcHook(DbApiHook):
+    """
+    Interact with Microsoft SQL Server using odbc.
+    """
+
+    conn_name_attr = 'mssql_conn_id'
+    default_conn_name = 'mssql_default'
+    supports_autocommit = True
+
+    def __init__(
+        self,
+        mssql_conn_id='mssql_default',
+        database=None,
+        schema=None,
+        driver=None,
+        dsn=None,
+        trx_isolation=pyodbc.SQL_TXN_READ_UNCOMMITTED,
+        autocommit=True,
+        *args,
+        **kwargs,
+    ):
+        super(MsSqlOdbcHook, self).__init__(*args, **kwargs)
+        self.mssql_conn_id = mssql_conn_id
+        self._database = database
+        self._driver = driver
+        self._dsn = dsn
+        self._conn_str = None
+        self.schema = schema
+        self.autocommit = autocommit
+        self._conn_extra = None
+        self._conn_extra_lower = None
+        self._conn = None
```

Comment:
I think this variable could be delete, IMO, this will be the only hook create mssql connect in Airflow, there is no where else could create it , so is could be delete, So as func `conn `

Reply:
i'm not sure what you mean?  the point of this is to store  the `Connection` object under the property `conn`, and that way repeated references to it will not require going back to metastore database to retrieve the conn again.

Reply:
Oh, I get your point ;-)